res/controllers/midi-components-0.0.js

```diff
@@ -146,18 +202,30 @@
         shiftChannel: false,
         shiftControl: false,
         send: function(value) {
-            if (this.midi === undefined || this.midi[0] === undefined || this.midi[1] === undefined) {
+            if (this.midi.out.length <= 0) {
                 return;
             }
-            midi.sendShortMsg(this.midi[0], this.midi[1], value);
+            for (const [midiStatus, midino] in this.midi.out) {
+                midi.sendShortMsg(midiStatus, midino, value);
+            }
+
             if (this.sendShifted) {
                 if (this.shiftChannel) {
-                    midi.sendShortMsg(this.midi[0] + this.shiftOffset, this.midi[1], value);
+                    midi.sendShortMsg(this.midi.out[0][0] + this.shiftOffset, this.midi.out[0][1], value);
                 } else if (this.shiftControl) {
-                    midi.sendShortMsg(this.midi[0], this.midi[1] + this.shiftOffset, value);
+                    midi.sendShortMsg(this.midi.out[0][0], this.midi.out[0][1] + this.shiftOffset, value);
                 }
             }
         },
+
+        get defaultInputHandler() {
+            return this.input;
+        },
```

Comment:
this is unnecessary IMO. JogWheelBasic should deal with its two input handlers by overwriting `connect` instead.